src/main/java/com/epam/training/designpatterns/fastfoodrestaurant/food/Condiment.java

```diff
@@ -0,0 +1,22 @@
+package com.epam.training.designpatterns.fastfoodrestaurant.food;
+
+public abstract class Condiment {
+	
+	protected Food baseFood;
+
+	protected Condiment(Food baseFood) {
+		this.baseFood = baseFood;
+	}
+	
+	protected Condiment() { }
```

Comment:
Akarunk paraméter nélküli konstruktort? A `baseFood` mindig kelleni fog (lásd null check kommentek), nem elég a fenti konstruktor? A setter miért kell?

Reply:
Valóban, ezt azért csináltam, mert amikor az Orderbe bele tettem eredetileg a condiment-eket (Ketchup, ...), akkor példányt adtam át new Ketchup() módszerrel (és ekkor kellett egy param nélküli konstruktor, és majd később a Chef settelte be a baseFood-ot). Ezt most lecseréltem olyan megoldásra, hogy az orderbe csak a class kerül be (Ketchup.class, Mustard.class), és majd a Chef dolga lesz példányosítani a baseFood-dal együtt